### PR TITLE
sidebar: scroll changed, background location match added

### PR DESCRIPTION
**Pull request by rcrdlbl**
see: #1185.

also see: #1202
we needed to also see if the background location was a match for being "in a group" to prevent modals that weren't underneath a group-specific route from launching the sidebar over to the main view.

### CODE REVIEW

**Comment by patosullivan** [comment]
Looking funky on mobile:
image

**Comment by patosullivan** [comment]
Also seeing a lot of extra space + unnecessary scrollbar when pinning just one group on desktop:
image

**Review by patosullivan — CHANGES_REQUESTED** [review]
Seeing some issues with the way this is rendering when a group is pinned on mobile or desktop.

**Comment by rcrdlbl** [comment]
@patosullivan can you check now? should be fixed

File: ui/src/nav/MobileRoot.tsx [hunk 1]
```diff
@@ -37,19 +37,21 @@ export default function MobileRoot() {
...
             <GroupList
               pinned
               groups={sortedGroups}
               pinnedGroups={sortedPinnedGroups}
             />
           </ul>
         ) : null}
-      </header>
-      <nav className="h-full flex-1 overflow-y-auto">
+        <li className="-mx-2 mt-5 grow border-t-2 border-gray-50 pt-3 pb-2">
```
Comment (arthyn) [inline ui/src/nav/MobileRoot.tsx:50]:
```suggestion
        <li className="-mx-2 sm:mt-5 grow border-t-2 border-gray-50 pt-3 pb-2">
```

File: ui/src/nav/MobileRoot.tsx [hunk 2]
```diff
@@ -37,19 +37,21 @@ export default function MobileRoot() {
           </Link>
         </div>
       </header>
-
-      <header className="flex-none px-2 py-1">
+      <nav className="h-full flex-1 overflow-y-auto overflow-x-hidden">
         {hasKeys(pinnedGroups) ? (
-          <ul className="mb-3 space-y-2 sm:mb-2 sm:space-y-0 md:mb-0">
+          <ul className="mb-3 space-y-2 px-2 sm:mb-2 sm:space-y-0 md:mb-0">
```
Comment (arthyn) [inline ui/src/nav/MobileRoot.tsx:42]: You added horizontal padding here but I think it needs go up one level to nav

**Review by arthyn — CHANGES_REQUESTED** [review]

**Comment by patosullivan** [comment]
Still seeing some weirdness:
image
image

**Comment by rcrdlbl** [comment]
@patosullivan are you sure you've pulled the latest commit? I'm unable to replicate the horizontal scrollbars or menu misalignment anymore on chrome mobile view or my macbook's iOS simulator
Screen Shot 2022-11-23 at 6 53 36 PM

**Review by patosullivan — APPROVED** [review]
Looks good! I must not have pulled last time.

**Pull request merged**